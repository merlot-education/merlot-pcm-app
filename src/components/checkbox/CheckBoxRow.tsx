import React from 'react'
import { View, StyleSheet, TouchableOpacity, Text, Image } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import Images from '../../assets'

import { ColorPallet, TextTheme } from '../../theme/theme'

interface Props {
  title: string
  accessibilityLabel?: string
  checked: boolean
  onPress: () => void
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    margin: 10,
  },
  text: {
    flexShrink: 1,
    ...TextTheme.normal,
    marginLeft: 10,
    color: ColorPallet.baseColors.black,
  },
  bottomIcon: {
    width: 60,
    height: 60,
  },
})

const CheckBoxRow: React.FC<Props> = ({
  title,
  accessibilityLabel,
  checked,
  onPress,
}) => {
  const accessible = !!(accessibilityLabel && accessibilityLabel !== '')

  return (
    <View style={styles.container} testID="checkBoxRowView">
      <TouchableOpacity
        style={styles.container}
        testID="checkBoxRow"
        accessible={accessible}
        accessibilityLabel={accessibilityLabel}
        onPress={onPress}
      >
        {checked ? (
          <Image source={Images.termsAcceptedIcon} style={styles.bottomIcon} />
        ) : (
          <Icon
            name="check-box-outline-blank"
            size={36}
            color={ColorPallet.brand.secondary}
          />
        )}
        <Text style={styles.text}>{title}</Text>
      </TouchableOpacity>
    </View>
  )
}

export default CheckBoxRow
