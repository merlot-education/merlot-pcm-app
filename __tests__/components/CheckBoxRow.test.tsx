import React from 'react'
import { create } from 'react-test-renderer'

import CheckBoxRow from '../../src/components/checkbox/CheckBoxRow'

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')

describe('CheckBoxRow Component', () => {
  it('CheckBoxRow renders correctly', () => {
    const tree = create(
      <CheckBoxRow title="CheckBoxRow" checked={false} />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
