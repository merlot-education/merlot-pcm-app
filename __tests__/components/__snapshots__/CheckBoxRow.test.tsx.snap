// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxRow Component CheckBoxRow renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <View
    accessible={false}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "margin": 10,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#1E9BF2"
      name="check-box-outline-blank"
      size={36}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "flexShrink": 1,
            "fontSize": 18,
            "fontWeight": "normal",
            "marginLeft": 10,
          },
        ]
      }
    >
      CheckBoxRow
    </Text>
  </View>
</View>
`;
